wandb/integration/catboost/catboost.py

```diff
@@ -0,0 +1,176 @@
+"""
+catboost init
+"""
+
+from pathlib import Path
+from types import SimpleNamespace
+from typing import Union
+
+from catboost import CatBoostClassifier, CatBoostRegressor  # type: ignore
+import wandb
+
+
+class WandbCallback:
+    """`WandbCallback` automatically integrates CatBoost with wandb.
+
+    Arguments:
+        - metric_period: (int) if you are passing `metric_period` to your CatBoost model please pass the same value here (default=1).
+
+    Passing `WandbCallback` to CatBoost will:
+    - log training and validation metrics at every `metric_period`
+    - log iteration at every `metric_period`
+
+    Example:
+        ```
+        train_pool = Pool(train[features], label=train["label"], cat_features=cat_features)
+        test_pool = Pool(test[features], label=test["label"], cat_features=cat_features)
+
+        model = CatBoostRegressor(
+            iterations=100,
+            loss_function="Cox",
+            eval_metric="Cox",
+        )
+
+        model.fit(
+            train_pool,
+            eval_set=test_pool,
+            callbacks=[WandbCallback()],
+        )
+        ```
+    """
+
+    def __init__(self, metric_period: int = 1):
+        if wandb.run is None:
+            raise wandb.Error("You must call `wandb.init()` before `WandbCallback()`")
+
+        self.metric_period: int = metric_period
+
+    def after_iteration(self, info: SimpleNamespace) -> bool:
+        if info.iteration % self.metric_period == 0:
+            for data, metric in info.metrics.items():
+                for metric_name, log in metric.items():
+                    wandb.log({f"{data}-{metric_name}": log[-1]}, commit=False)
+
+            wandb.log({f"iteration@metric-period-{self.metric_period}": info.iteration})
+
+        return True
+
+
+def _checkpoint_artifact(
+    model: Union[CatBoostClassifier, CatBoostRegressor], aliases: list
+) -> None:
+    """
+    Upload model checkpoint as W&B artifact
+    """
+    if wandb.run is None:
+        raise wandb.Error(
+            "You must call `wandb.init()` before `_checkpoint_artifact()`"
+        )
+
+    model_name = f"model_{wandb.run.id}"
+    # save the model in the default `cbm` format
+    model_path = Path(wandb.run.dir) / "model"
+
+    model.save_model(model_path)
+
+    model_artifact = wandb.Artifact(name=model_name, type="model")
+    model_artifact.add_file(str(model_path))
+    wandb.log_artifact(model_artifact, aliases=aliases)
```

Comment:
probably here we also should use the internal log_artifact: `wandb.run._log_artifact`

Reply:
also todo?

Reply:
no... we actually have this function :)))
https://github.com/wandb/client/blob/55b885c14018677b916e006ca9dbc9814db99d7c/wandb/sdk/wandb_run.py#L2165

Reply:
Oops. What's your preference? I don't care :) Just generally prefer using public interfaces ;)

Reply:
hmmm... even for integration? I'm not sure... 
Just raising this as @raubitsj suggested using internal interface in my PR

Reply:
In this case there isnt a reason now to prefer _log_artifact over log_artifact. 
If it was using some weird feature of log_artifact and we wanted capture that in a telemetry object then using the internal one would be better.. but this looks pretty standard, it is fine how it is.